lib/util.js

```diff
@@ -64,33 +60,33 @@ export function viewport(obj, options) {
  * @api private
  */
 
-export function save(streams) {
+export async function save(streams) {
 	const files = [];
 
-	function end() {
-		return Promise.all(files.map(file => pify(rimraf, Promise)(file)));
+	async function end() {
+		return await* files.map(file => pify(rimraf, Promise)(file));
 	}
 
-	process.on('SIGINT', () => end().then(process.exit));
+	process.on('SIGINT', async () => process.exit(await end()));
 
-	return Promise.all(streams.map(stream => {
-		return pify(mkdirp, Promise)(this.dest()).then(() => {
-			return new Promise((resolve, reject) => {
-				const dest = path.join(this.dest(), stream.filename);
-				const write = fsWriteStreamAtomic(dest);
+	return await* streams.map(async stream => {
+		return new Promise(async (resolve, reject) => {
```

Comment:
Because this way it creates a list of promises first and executes the list in parallel afterwards (`await*`). If you don't wrap it, everything gets executed in series.

Although that is how I think it works. If you see any improvements that we could make, would be happy to implement. Not very satisfied about this piece, but not sure if it can be written better.

I guess the `streams.map(async stream => {` is a typo and should be `streams.map(stream =>`.